### ISSUE TITLE
HelmChart api object - comma problem

### ISSUE THREAD

**Issue by Earsil** [k3s-io/helm-controller#94]
Hi,
I am using k3s cluster version v1.20.4+k3s1 and I am trying to apply a rancher HelmChart using this controller. It works, however I have a problem when I try to use a comma value in set. I am aware that helm requires to escape a comma sign, but I guess somehow when helm-install starts, it does something weird to this setting.

My Helm chart object looks like this:

```apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  creationTimestamp: "2021-04-22T09:47:03Z"
  finalizers:
  - wrangler.cattle.io/helm-controller
  generation: 3
  name: rancher
  namespace: kube-system
  resourceVersion: "2177"
  uid: bd42926e-355a-49d6-ab28-b15331067363
spec:
  chart: rancher
  repo: https://releases.rancher.com/server-charts/stable
  set:
    hostname: rancher-test.com
    http_proxy: http://example-wproxy.test.com:8080
    https_proxy: https://example-wproxy.test.com:8443
    ingress.tls.source: secret
    no_proxy: 127.0.0.0/8\\,10.0.1.0/8
  targetNamespace: cattle-system
status:
  jobName: helm-install-rancher
```

and helm-install logs look like this:
```
helm_v3 install --namespace cattle-system --repo https://releases.rancher.com/server-charts/stable --set-string hostname=rancher-test.com --set-string http_proxy=http://example-wproxy.test.com:8080 --set-string https_proxy=https://example-wproxy.test.com:8443 --set-string ingress.tls.source=secret --set-string rancher rancher
Error: must either provide a name or specify --generate-name
```

From job logs it seems that no_proxy part is missing, there is nothing atfer last --set-string

And when I don’t escape comma i get an error about it:
```
helm_v3 install --namespace cattle-system --repo https://releases.rancher.com/server-charts/stable --set-string hostname=rancher-test.com --set-string http_proxy=http://example-wproxy.test.com:8080 --set-string https_proxy=https://example-wproxy.test.com:8443 --set-string ingress.tls.source=secret --set-string no_proxy=127.0.0.0/8,10.0.1.0/8 rancher rancher

Error: failed parsing --set-string data: key map "10" has no value
```

I also tried adding a different amount of \ and other " and more more, unfortunately with no success.

Is this a bug or is there maybe a workaround for this problem?

**Comment by brandond** [comment]
This should be resolved by #95, which won't land in K3s and RKE2 for a bit, but you can fix it on current versions by overriding the default `spec.jobImage` and manually escaping the comma in the set string:
```yaml
kind: HelmChart
metadata:
  name: rancher
  namespace: kube-system
spec:
  jobImage: rancher/klipper-helm:v0.5.0-build20210505
  chart: rancher
  repo: https://releases.rancher.com/server-charts/stable
  set:
    hostname: rancher-test.com
    http_proxy: http://example-wproxy.test.com:8080
    https_proxy: https://example-wproxy.test.com:8443
    ingress.tls.source: secret
    no_proxy: 127.0.0.0/8\,10.0.1.0/8
  targetNamespace: cattle-system
```

**Comment by Earsil** [comment]
Thanks, so it was a bug. 
As a workaround I just ran the helm manually, but I guess editing job will work too.
